sphinx/ext/autosummary/__init__.py

```diff
@@ -686,7 +690,9 @@ def import_ivar_by_name(name: str, prefixes: List[str] = [None]) -> Tuple[str, A
         # check for presence in `annotations` to include dataclass attributes
         if (qualname, attr) in analyzer.attr_docs or (qualname, attr) in analyzer.annotations:
             return real_name + "." + attr, INSTANCEATTR, obj, modname
-    except (ImportError, ValueError, PycodeError):
+    except (ImportError, ValueError, PycodeError) as e:
+        if last_errors is not None:
+            last_errors.append(str(e.args[0]))
```

Comment:
```suggestion
            last_errors.append(str(e))
```

Here I’m not so sure: What’s `PycodeError`’s `__str__` method?

Reply:
Jep, in this case it's a bit suboptimal. We can split it up into two excepts, one for ImportError and ValueError, which uses `str(e.args[0])` the second one just uses `str(e)` for PycodeError, since there internally it is already accessing args[0]:
https://github.com/sphinx-doc/sphinx/blob/fe47bf46cbb6a615b01d6c5dc9fdcac58c940f69/sphinx/errors.py#L118